check-submodule.py

```diff
@@ -229,6 +210,82 @@ def save_diff_files(
     asyncio.run(json_diff.diff())
 
 
+# Run rust-code-analysis on the chosen repository to compute metrics.
+def run_rca(
```

Comment:
Please move this function where it was before, so the diff is smaller and simpler to review